From 53ede2b6fccd1abd639579c5962b497ead2b1359 Mon Sep 17 00:00:00 2001
From: zhangmeng <zhangmeng.kevin@spacemit.com>
Date: Tue, 19 Dec 2023 22:17:20 +0800
Subject: [PATCH 0202/1189] update dtsi for k1-x platform

1. add sscofpmf feature for cpu
2. fix machine-mode external interrupt number configuration

Change-Id: I5558aa12cc3251616548dbd5c86e39bbaab24390
---
 arch/riscv/boot/dts/spacemit/k1-x.dtsi | 36 ++++++++++++--------------
 1 file changed, 17 insertions(+), 19 deletions(-)

diff --git a/arch/riscv/boot/dts/spacemit/k1-x.dtsi b/arch/riscv/boot/dts/spacemit/k1-x.dtsi
index af80d7f78630..fc05c66d3b99 100644
--- a/arch/riscv/boot/dts/spacemit/k1-x.dtsi
+++ b/arch/riscv/boot/dts/spacemit/k1-x.dtsi
@@ -10,7 +10,7 @@
 #include <dt-bindings/display/spacemit-dpu.h>
 
 / {
-	compatible = "spacemit,k1-x", "riscv";
+	compatible = "spacemit,k1-x";
 	#address-cells = <2>;
 	#size-cells = <2>;
 
@@ -35,7 +35,7 @@ cpu_0: cpu@0 {
 			device_type = "cpu";
 			reg = <0>;
 			status = "okay";
-			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc";
+			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc_sscofpmf";
 			riscv,cbom-block-size = <64>;
 			i-cache-block-size = <64>;
 			i-cache-size = <32768>;
@@ -61,7 +61,7 @@ cpu_1: cpu@1 {
 			reg = <1>;
 			status = "okay";
 			compatible = "riscv";
-			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc";
+			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc_sscofpmf";
 			riscv,cbom-block-size = <64>;
 			i-cache-block-size = <64>;
 			i-cache-size = <32768>;
@@ -87,7 +87,7 @@ cpu_2: cpu@2 {
 			reg = <2>;
 			status = "okay";
 			compatible = "riscv";
-			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc";
+			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc_sscofpmf";
 			riscv,cbom-block-size = <64>;
 			i-cache-block-size = <64>;
 			i-cache-size = <32768>;
@@ -113,7 +113,7 @@ cpu_3: cpu@3 {
 			reg = <3>;
 			status = "okay";
 			compatible = "riscv";
-			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc";
+			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc_sscofpmf";
 			riscv,cbom-block-size = <64>;
 			i-cache-block-size = <64>;
 			i-cache-size = <32768>;
@@ -139,7 +139,7 @@ cpu_4: cpu@4 {
 			reg = <4>;
 			status = "okay";
 			compatible = "riscv";
-			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc";
+			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc_sscofpmf";
 			riscv,cbom-block-size = <64>;
 			i-cache-block-size = <64>;
 			i-cache-size = <32768>;
@@ -165,7 +165,7 @@ cpu_5: cpu@5 {
 			reg = <5>;
 			status = "okay";
 			compatible = "riscv";
-			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc";
+			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc_sscofpmf";
 			riscv,cbom-block-size = <64>;
 			i-cache-block-size = <64>;
 			i-cache-size = <32768>;
@@ -191,7 +191,7 @@ cpu_6: cpu@6 {
 			reg = <6>;
 			status = "okay";
 			compatible = "riscv";
-			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc";
+			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc_sscofpmf";
 			riscv,cbom-block-size = <64>;
 			i-cache-block-size = <64>;
 			i-cache-size = <32768>;
@@ -217,7 +217,7 @@ cpu_7: cpu@7 {
 			reg = <7>;
 			status = "okay";
 			compatible = "riscv";
-			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc";
+			riscv,isa = "rv64imafdcvsu_zicsr_zifencei_zicbom_zihintpause_zba_zbb_zbc_zbs_svpbmt_sstc_sscofpmf";
 			riscv,cbom-block-size = <64>;
 			i-cache-block-size = <64>;
 			i-cache-size = <32768>;
@@ -442,10 +442,8 @@ &cpu4_intc  3 &cpu4_intc  7
 				&cpu5_intc  3 &cpu5_intc  7
 				&cpu6_intc  3 &cpu6_intc  7
 				&cpu7_intc  3 &cpu7_intc  7
-
 			>;
 			reg = <0x0 0xE4000000 0x0 0x00010000>;
-			clint,has-no-64bit-mmio;
 		};
 
 		ccu: clock-controller@d4050000 {
@@ -485,14 +483,14 @@ intc: interrupt-controller@e0000000 {
 			compatible = "riscv,plic0";
 			interrupt-controller;
 			interrupts-extended = <
-				&cpu0_intc 0xffffffff &cpu0_intc 9
-				&cpu1_intc 0xffffffff &cpu1_intc 9
-				&cpu2_intc 0xffffffff &cpu2_intc 9
-				&cpu3_intc 0xffffffff &cpu3_intc 9
-				&cpu4_intc 0xffffffff &cpu4_intc 9
-				&cpu5_intc 0xffffffff &cpu5_intc 9
-				&cpu6_intc 0xffffffff &cpu6_intc 9
-				&cpu7_intc 0xffffffff &cpu7_intc 9
+				&cpu0_intc 11 &cpu0_intc 9
+				&cpu1_intc 11 &cpu1_intc 9
+				&cpu2_intc 11 &cpu2_intc 9
+				&cpu3_intc 11 &cpu3_intc 9
+				&cpu4_intc 11 &cpu4_intc 9
+				&cpu5_intc 11 &cpu5_intc 9
+				&cpu6_intc 11 &cpu6_intc 9
+				&cpu7_intc 11 &cpu7_intc 9
 			>;
 			reg = <0x0 0xE0000000 0x0 0x04000000>;
 			reg-names = "control";
-- 
2.47.1

